extensions/amp-social-share/0.2/social-share.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as CSS from './social-share.css';
+import * as Preact from '../../../src/preact';
+import {Keys} from '../../../src/utils/key-codes';
+import {SocialShareIcon} from '../../../third_party/optimized-svg-icons/social-share-svgs';
+import {dev, devAssert, userAssert} from '../../../src/log';
+import {getAmpContext} from '../../../src/preact/context';
+import {openWindowDialog} from '../../../src/dom';
+import {parseQueryString} from '../../../src/url';
+import {startsWith} from '../../../src/string';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+const DEFAULT_WIDTH = 60;
+const DEFAULT_HEIGHT = 44;
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function SocialShare(props) {
+  const context = Preact.useContext(getAmpContext());
+
+  /**
+   * Handle key presses on the element.
+   * @param {!Event} event
+   * @private
+   */
+  function handleKeyPress(event) {
+    const {key} = event;
+    if (key == Keys.SPACE || key == Keys.ENTER) {
+      event.preventDefault();
+      handleActivation();
+    }
+  }
+
+  /** @private */
+  function handleActivation() {
+    const href = context['href'] || props['href'];
+    const target = context['target'] || props['target'] || '_blank';
+    userAssert(href && target, 'Clicked before href is set.');
+    dev().assertString(href);
```

Comment:
/cc @dvoytenko. I think we can clean this kind of assertion up with the props typedefs. But, do we want to support the dev experience in React mode? That would mean we should keep the asserts here.

Reply:
I agree: we need PropTypes or similar on React side. Any `dev().assert` style of assertion should be limited to AMP side only.

Reply:
I imported `PropTypes` in the recent change though the error for mismatched types appears to be a `console.error` that's not fatal. If we want to replace `dev().assert` exactly we may want to use something else.

Reply:
We can only import `PropTypes` once we ensure they are stripped by our compiler. So @jridgewell should probably advise on this. But I do think we want to declare prop types one way or the other and I've only been doing PropTypes so far myself.